Update aiohttp dependency

agentultra opened the pull request:
This change updates aiohttp to >= 0.22.1 and updates the code to the new
APIs.

agentultra committed:
Update aiohttp dependency

This change updates aiohttp to >= 0.22.1 and updates the code to the new
APIs.

asvetlov commented on aiocouchdb/client.py:
```diff
@@ -172,7 +172,7 @@ def request(method, url, *,
 
 
-class HttpRequest(aiohttp.client.ClientRequest):
+class HttpRequest(aiohttp.client_reqrep.ClientRequest):
```
please use just `aiohttp.ClientRequest`

asvetlov commented on aiocouchdb/client.py:
```diff
@@ -209,7 +209,7 @@ def update_path(self, params):
 
 
-class HttpResponse(aiohttp.client.ClientResponse):
+class HttpResponse(aiohttp.client_reqrep.ClientResponse):
```
The same

asvetlov commented on aiocouchdb/tests/test_feeds.py:
```diff
@@ -108,19 +108,16 @@ def test_buffer_workflow(self):
         self.assertTrue(feed.is_active())
 
         result = yield from feed.next()
-        self.assertEqual(feed._queue.qsize(), buf_size)
```
Please describe why you have removed the check?

kxepal replied:
Probably, because it causes test failure. Whole this test is sort of flaky with new Python/asyncio versions.

agentultra replied:
It was testing internal state of the object; `asyncio.Queue` is well tested by it's own library and its correctness shouldn't be a concern here, should it?

kxepal commented:
@agentultra Hi
Thanks for PR, but suddenly it requires a little bit more work to do. It's too long to explain, just hold on a bit. I hope to return to the project in a couple of days.

weeverjames committed:
Remove unused import

agentultra commented:
@kxepal Understood... some of the APIs have changed quite a bit and I did my best on the first pass to keep things lined up.

I'm working on a script that uses aiocouchdb with the aiobotocore library and they use conflicting versions of aiohttp. I'm here to help any way I can as I really need to get this working.

weeverjames committed:
Revert "Remove unused import"

This reverts commit df9ac9af93bef0bc677a1ac590ac883984e17705.

weeverjames committed:
Remove unused import and change HttpRequest parent

Following up on some review comments I changed the HttpRequest and
HttpResponse classes to subclass from aiohttp.ClientRequest and
aiohttp.ClientResponse respectively.

The lint check was catching an unused import in hdrs.py which I added a
skip to since it's importing names into the module that are used
elsewhere.

SakuraSound commented:
Any chance of this getting done?

kxepal commented:
Oh, sorry! Yes, will handle this today.

SakuraSound commented:
Thank you! I was originally going to fork and try to push something, but i saw that this was one Error shy of being merged.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/aio-libs/aiocouchdb?pullRequest=16) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/aio-libs/aiocouchdb?pullRequest=16) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: agentultra<br/>:x: weeverjames<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/aio-libs/aiocouchdb?pullRequest=16) it.</sub>
